### PR TITLE
Enhancement/delete promotions

### PR DESCRIPTION
**Pull request by anirudh28**
Covers all tests for the delete feature.

### CODE REVIEW

File: service/routes.py [hunk 1]
```diff
@@ -97,6 +97,21 @@ def read(promotion_id):
...
+def delete(promotion_id):
+    """Deletes a Promotion with promotion_id with the fields included in the body of the request"""
+    app.logger.info(f"Got request to delete Promotion with id: {promotion_id}")
+    promotion = Promotion.find(promotion_id)
+    if not promotion:
+        abort_with_error(status.HTTP_404_NOT_FOUND,
+                         f"Promotion with id: {promotion_id} not found")
+    try:
+        promotion.delete()
+        return jsonify({}), status.HTTP_200_OK
```
Comment (mingyi850) [inline service/routes.py:111]: I think for successful delete we should return 204 instead of 200

File: service/routes.py [hunk 2]
```diff
@@ -97,6 +97,21 @@ def read(promotion_id):
...
+    except Exception as error:  # pylint: disable=broad-except
+        return abort_with_error(status.HTTP_500_INTERNAL_SERVER_ERROR, f"An error occurred : {error}")
```
Comment (mingyi850) [inline service/routes.py:113]: I think we can remove this and leave it to the error handler instead - i updated this in the newest PR you can reference

**Review by mingyi850 — CHANGES_REQUESTED** [review]
Tests look good! Just some small changes which need to be made

**Comment by mingyi850** [comment]
Oh also, please link the PR to the issue too for zenhub

File: service/routes.py (hunk 2 above)
Comment (anirudh28) [reply service/routes.py:113]: I should delete the test which was created to catch this too right?

File: service/routes.py (hunk 1 above)
Comment (anirudh28) [reply service/routes.py:111]: Done.

File: service/routes.py (hunk 2 above)
Comment (anirudh28) [reply service/routes.py]: Done.

**Review by mingyi850 — APPROVED** [review]
LGTM

**Pull request merged**